test/docker-e2e/e2e_test.go

```diff
@@ -0,0 +1,157 @@
+package docker_e2e
+
+import (
+	"context"
+	"fmt"
+	"github.com/celestiaorg/celestia-app/v4/app"
+	"github.com/celestiaorg/go-square/v2/share"
+	celestiadockertypes "github.com/chatton/celestia-test/framework/docker"
+	"github.com/chatton/celestia-test/framework/testutil/maps"
+	celestiatypes "github.com/chatton/celestia-test/framework/types"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/moby/moby/client"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	multiplexerImage   = "ghcr.io/celestiaorg/celestia-app"
+	txsimImage         = "ghcr.io/celestiaorg/txsim"
+	defaultCelestiaTag = "v4.0.0-rc4"
+)
+
+func TestCelestiaTestSuite(t *testing.T) {
+	suite.Run(t, new(CelestiaTestSuite))
+}
+
+type CelestiaTestSuite struct {
+	suite.Suite
+	logger  *zap.Logger
+	client  *client.Client
+	network string
+}
+
+func (s *CelestiaTestSuite) SetupSuite() {
+	s.logger = zaptest.NewLogger(s.T())
+	s.logger.Info("Setting up Celestia test suite")
+	s.client, s.network = celestiadockertypes.DockerSetup(s.T())
+}
+
+func (s *CelestiaTestSuite) CreateCelestiaChainProvider(appVersion string) celestiatypes.ChainProvider {
+	numValidators := 4
+	numFullNodes := 0
+
+	enc := testutil.MakeTestEncodingConfig(app.ModuleEncodingRegisters...)
+
+	cfg := celestiadockertypes.Config{
+		Logger:          s.logger,
+		DockerClient:    s.client,
+		DockerNetworkID: s.network,
+		Type:            "cosmos",
+		Name:            "celestia",
+		Version:         getCelestiaTag(),
+		NumValidators:   &numValidators,
+		NumFullNodes:    &numFullNodes,
+		ChainID:         "celestia",
+		Images: []celestiadockertypes.DockerImage{
+			{
+				Repository: multiplexerImage,
+				Version:    getCelestiaTag(),
+				UIDGID:     "10001:10001",
+			},
+		},
+		Bin:           "celestia-appd",
+		Bech32Prefix:  "celestia",
+		Denom:         "utia",
+		CoinType:      "118",
+		GasPrices:     "0.025utia",
+		GasAdjustment: 1.3,
+		ModifyGenesis: func(config celestiadockertypes.Config, bytes []byte) ([]byte, error) {
```

Comment:
[not blocking]
does this mean that the genesis is defined in the `celestiadockertypes` repo?

Reply:
yes, this is basically the type that is being created right here. It's passed as an arg to this function in case any changes to the genesis file are dependant on version, image etc.